pkgs/development/interpreters/python/python-packages-base.nix

```diff
@@ -17,11 +17,16 @@ let
   makeOverridablePythonPackage = f: origArgs:
     let
       ff = f origArgs;
+      args = lib.fix (lib.extends
+        (_: previousAttrs: {
+          passthru = (previousAttrs.passthru or { }) // {
+            overridePythonAttrs = newArgs: makeOverridablePythonPackage f (overrideWith newArgs);
+          };
+        })
+        (_: origArgs));
```

Comment:
is this correct @roberth ?

Reply:
I think the ordering of the `extends` arguments is ok. "Self" is ignored, so I think the types line up (but I didn't prove it).
Looks about right.

Reply:
Performance has improved by 2-3%, judging by memory stats [before] and [after] "use extends instead of overrideAttrs".
Not enough to make python changes as cheap as the other changes though.
I wonder how much change/noise is caused by `master`, which I believe is merged into the feature branch before ofborg evaluates.

[before]: https://github.com/NixOS/nixpkgs/runs/10147970728
[after]: https://github.com/NixOS/nixpkgs/runs/10161222989

Reply:
rebased on master again

Reply:
![image](https://user-images.githubusercontent.com/56650223/208271036-9719c4ca-817f-4ef0-8986-d4c9792f870b.png)

Reply:
We should probably bench this manually anyway, since cross is not part of ofborg's eval I think.

Reply:
![image](https://user-images.githubusercontent.com/56650223/208271773-6d9bab65-626e-4cf2-a043-1d455d7c33ed.png)

Reply:
![image](https://user-images.githubusercontent.com/56650223/208322610-f574aa3e-203b-4509-a4aa-e0971c5c827a.png)

Reply:
current stats ![image](https://user-images.githubusercontent.com/56650223/212521945-0d8d3b02-acf3-4a00-b241-c531eb46c6db.png)